tasks/test.py

```diff
@@ -33,26 +34,55 @@ def ensure_bytes(s):
     return s
 
 
+@contextmanager
+def environ(env):
+    original_environ = os.environ.copy()
+    os.environ.update(env)
+    yield
+    for var in env.keys():
+        if var in original_environ:
+            os.environ[var] = original_environ[var]
+        else:
+            os.environ.pop(var)
+
+
 TOOL_LIST = [
     'github.com/client9/misspell/cmd/misspell',
     'github.com/frapposelli/wwhrd',
     'github.com/fzipp/gocyclo',
     'github.com/go-enry/go-license-detector/v4/cmd/license-detector',
-    'github.com/golangci/golangci-lint/cmd/golangci-lint',
     'github.com/gordonklaus/ineffassign',
     'github.com/goware/modvendor',
     'github.com/mgechev/revive',
+    'github.com/stormcat24/protodep',
+    'golang.org/x/lint/golint',
     'gotest.tools/gotestsum',
     'honnef.co/go/tools/cmd/staticcheck',
 ]
 
+TOOL_LIST_GET = [
+    'github.com/golangci/golangci-lint/cmd/golangci-lint@v1.40.1',
+    'github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway@v1.12.2',
+    'github.com/golang/protobuf/protoc-gen-go@v1.3.2',
+    'github.com/golang/mock/mockgen@v1.5.0',
+]
+
 
 @task
 def install_tools(ctx):
     """Install all Go tools for testing."""
-    with ctx.cd("internal/tools"):
-        for tool in TOOL_LIST:
-            ctx.run("go install {}".format(tool))
+    with environ({'GO111MODULE': 'on'}):
```

Comment:
Why do we need this?

Reply:
Well, because the default value for that env var varies across go versions, it can also be left behind if the user messes with the environment, so I figured this was the right thing to do. I wish `invoke` had a nicer way to address this.